### PR TITLE
feat: Run examples from any package

### PR DESCRIPTION
**Pull request by gadenbuie**
This PR makes it possible for any package to provide shiny examples in `inst/examples-shiny` and have them be runnable/discoverable via `runExample()`.

This PR simply exposes `package` in the `runExample()` function signature, giving us the core feature described in #3907. The work in #3917 is still useful to make this feature more user-friendly and can be done in the next release. But we can get the core feature out in the wild so people can start to use it.

Note that for non-shiny packages, the examples directory must be `inst/examples-shiny`. The `shiny.legacy.examples` option only has takes effect when looking for examples in shiny itself.

Note: stacks on top of #4003. We can merge this into #4003 so we can merge the examples changes all at once.

### CODE REVIEW

File: R/runapp.R [hunk 1]
```diff
@@ -462,34 +472,41 @@ stopApp <- function(returnValue = invisible()) {
...
+  examplesDir <- system_file(
+    if (use_legacy_shiny_examples) "examples" else "examples-shiny",
+    package = package
+  )
+
   dir <- resolve(examplesDir, example)
+
   if (is.null(dir)) {
+    valid_examples <- sprintf(
+      'Valid examples are "%s"',
```
Comment (schloerke) [inline R/runapp.R:496]:
```suggestion
      'Valid examples: "%s"',
```

File: R/runapp.R [hunk 2]
```diff
@@ -462,34 +472,41 @@ stopApp <- function(returnValue = invisible()) {
...
+      paste(list.files(examplesDir), collapse = '", "')
```
Comment (schloerke) [inline R/runapp.R:497]:
```suggestion
      package,
      paste(list.files(examplesDir), collapse = '", "')
```

File: R/runapp.R [hunk 3]
```diff
@@ -462,34 +472,41 @@ stopApp <- function(returnValue = invisible()) {
...
+    isTRUE(getOption('shiny.legacy.examples', FALSE))
+
+  examplesDir <- system_file(
+    if (use_legacy_shiny_examples) "examples" else "examples-shiny",
+    package = package
+  )
+
   dir <- resolve(examplesDir, example)
+
   if (is.null(dir)) {
```
Comment (schloerke) [inline R/runapp.R:494]: Maybe add check for missing package?

**Review by schloerke — APPROVED** [review]

File: R/runapp.R (hunk 3 above)
Comment (gadenbuie) [reply R/runapp.R:498]: Good call, thanks!

**Review by cpsievert — APPROVED** [review]

**Pull request merged**